### PR TITLE
Add simple installer fixes #48

### LINKED ISSUE #48: Add installer source code/migrate to non commercial free installer generator
Hello, does someone know where the source code for the windows installer can be found?

I noticed that the installer was generated with installsimple, a commercial product.
Would be be possible to use something freely available instead?

Some possible options are:
* Innosetup: https://jrsoftware.org/isinfo.php
* NSIS: https://nsis.sourceforge.io/Download
* WiX Toolset: https://wixtoolset.org/

### LibreSprite and System version

* LibreSprite (latest):
* System (Windows):

### PR DESCRIPTION
**Pull request by Shoozza**
This includes a simple innosetup based installer.

### CODE REVIEW

**Comment by PerryHugh** [comment]
I don't use Windows, so someone else will have to test it.

**Comment by MintStudios** [comment]
@Shoozza So, can you compile and have the executable on hand? I see the script, just use a website like this https://anonfile.com/
To temporarily host it so I can see it.

File: installer/installer.iss [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#define MyAppVersion "1.0.0-dev"
+#define MyAppPublisher "LibreSprite Team"
+#define MyAppURL "https://libresprite.github.io/"
+#define MyAppExeName "libresprite.exe"
+
+[Setup]
+; NOTE: The value of AppId uniquely identifies this application.
+; Do not use the same AppId value in installers for other applications.
+; (To generate a new GUID, click Tools | Generate GUID inside the IDE.)
+AppId={{8FC54751-06C8-4BCC-9D83-CC241E875560}
```
Comment (Ev1lbl0w) [inline installer/installer.iss:14]: Double `{`?

**Comment by Shoozza** [comment]
Yeah I uploaded the installer: https://anonfiles.com/V2R1ccEao6/LibreSpriteInstaller_exe
Btw if you want to compile your own you just need to download and install innosetup.
I used innosetup-qsp-6.0.5 (the quick start pack) but the normal one should work as well.

**Comment by jjhaggar** [comment]
@Shoozza I have just tested it on Windows 7 ultimate 64 bits and it works fine. 
The only problem is that you must right click & run the installer as administrator or you'll get this error: 
imagen
(yes, it's Windows 7, I just like the classic 98 theme XD)

**Comment by Shoozza** [comment]
Could it be that your antivirus does something?
https://stackoverflow.com/questions/45754816/innosetup-fails-unable-to-execute-file-in-the-temporary-directory-setup-abor

**Comment by jjhaggar** [comment]
Maybe, but I just have Windows Defender installed, could that software be the problem?

**Comment by Shoozza** [comment]
Yeah, maybe you could try if the issue still happens if it's disabled?

**Comment by jjhaggar** [comment]
I have just tried disabling it and restarting the computer but the error still happens.

**Comment by Shoozza** [comment]
I tried to find a way to make it run as admin by default but there isn't a way.
We could modify the manifest (but that has also a downside of breaking the signature).
Seems like inno setup elevates to admin at runtime which for some reason isn't working on your windows 7 version (reasons can be antivirus or something else).
https://stackoverflow.com/questions/14499039/inno-setup-and-the-windows-uac-shield

Maybe it's not a big deal (windows 7 is EOL anyways)?

**Comment by jjhaggar** [comment]
If we warn the users about the error and tell them about method to install it in case of that error arising (a warning in the download page, or in a FAQ for example) I think there shouldn't be any problems.

Also, I'd like to ask one thing, could we also distribute the Windows binaries without installer? (I mean just a zip with the files) I don't know if the program works as a portable one as it is now, but if that's possible I think a lot of people would like havind that option too.

**Comment by Ev1lbl0w** [comment]
> Also, I'd like to ask one thing, could we also distribute the Windows binaries without installer? (I mean just a zip with the files) I don't know if the program works as a portable one as it is now, but if that's possible I think a lot of people would like having that option too.

This should work in theory already; the installer, from what I understand, does nothing more than just copying files. I agree we should offer a "portable" alternative.

**Comment by Shoozza** [comment]
Yeah currently it's mostly unpacking (also register uninstaller in "programs and features" and add shortcut on desktop and to startmenu)
But things like register file associations, side by side installation, portable installation could be added later.

**Comment by Shoozza** [comment]
You can replace the header icon (top right side), make the background of the dialog white and replace the finish page left image.
Is that what you meant?

https://i.imgur.com/6VMNvKd.png
https://i.imgur.com/d0PC0Hc.png

Note both are bmp files and you can find them under 'C:\Program Files (x86)\Inno Setup 6'

**Comment by MintStudios** [comment]
I'll merge this, as this is OK for now. We should put the binary in the releases section. After a few more days, after Spanish translations get merged and we can compile for windows in Travis CI, I think we should create a new release. @PerryHugh I personally think that the version name should be 1.0.1, since this doesn't change the LibreSprite code itself. But would 1.1 be more appropriate?

**Comment by Ev1lbl0w** [comment]
I vote for the [semver](https://semver.org/) versioning system. Since all these changes haven't introduced new features and neither "imcompatibility issues" with the last version, the rules say we upgrade to 1.0.1.

However, for a fork that had no development for over 4 years, I think a 1.1 would be more appropriate 😄

**Comment by PerryHugh** [comment]
I think the best method is the fairly standard way of release branches.

When we move into making a release, we would freeze master in a new branch titled for whichever release version (1.0, 1.1?) and package that for testing and finally release. This enables master to maintain a version-free state.

**Pull request merged**